client/dashboard/sites-ciab/index.tsx

```diff
@@ -60,24 +53,23 @@ export default function CIABSites() {
 	const navigate = useNavigate( { from: sitesRoute.fullPath } );
 	const queryClient = useQueryClient();
 	const currentSearchParams = sitesRoute.useSearch();
-	const viewSearchParams: ViewSearchParams = currentSearchParams.view ?? {};
 	const isRestoringAccount = !! currentSearchParams.restored;
 
 	const { user } = useAuth();
 	const { queries } = useAppContext();
 	const { setShowHelpCenter } = useHelpCenter();
 	const { data: isAutomattician } = useSuspenseQuery( isAutomatticianQuery() );
-	const { data: viewPreferences } = useSuspenseQuery( userPreferenceQuery( 'ciab-sites-view' ) );
-	const { mutate: updateViewPreferences } = useMutation(
-		userPreferenceMutation( 'ciab-sites-view' )
-	);
 
-	const { defaultView, view } = getView( {
+	const defaultView = getDefaultView( {
 		user,
 		isAutomattician,
 		isRestoringAccount,
-		viewPreferences,
-		viewSearchParams,
+	} );
+
+	const { view, updateView } = useView( {
+		slug: 'sites-ciab',
+		defaultView,
+		defaultFields: DEFAULT_FIELDS,
```

Comment:
Is it possible to pass it to `getDefaultView` function? Then, the `defaultFields` could be replaced by `defaultView.fields`?

Reply:
It seems to work well, even if some fields are later filtered out. However, we still need to address the circular dependency issue, since not all use cases define DEFAULT_FIELDS. If we aren’t able to resolve this, then we’ll need to define the default fields in every place first, and then call the `getFields` (or `useFields`) function later if the logic depends on the view 🥲

Reply:
Brilliant idea: what if we sanitize the view in the DataViews wrapper (that I'm going to write)? So `useView()` only needs `slug` and `defaultView`. Then, the wrapper sanitizes the view based on the passed `view` and `fields` 🤯

Reply:
Maybe we just need to propose changes to [here](https://github.com/WordPress/gutenberg/blob/a0f4a6e3ce78a1acea47daa6b97e0414e44e0ba2/packages/dataviews/src/utils/filter-sort-and-paginate.ts#L118) to check `enableSorting` and then we can get rid of the dependency of the `defaultFields` 🤔